### PR TITLE
Updated nightly versions to CSM 1.12 image tags

### PR DESCRIPTION
**Pull request by rajendraindukuri**
# Description
- Updated nightly versions to CSM 1.12 image tags in OperatorConfig folder

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1435|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
- Made sure all the unit tests are passing
- E2E sanity and pflex tests run and passing. Results tracked internally.
- ran unity e2e, results tracked internally 
- ran pscale e2e: ``` nohup ./run-e2e-test.sh --pscale &```, results tracked internally 
- ran pstore e2e: ``` nohup ./run-e2e-test.sh --pstore &```, results tracked internally 
- ran pmax e2e: ``` nohup ./run-e2e-test.sh --pmax & ```, results tracked internally 
- ran auth v2 e2e w/ pflex to verify everything worked

### CODE REVIEW

File: bundle/manifests/dell-csm-operator.clusterserviceversion.yaml
Comment (hoppea2) [inline bundle/manifests/dell-csm-operator.clusterserviceversion.yaml:17]: These are ANK8S components and should be removed

File: bundle/manifests/dell-csm-operator.clusterserviceversion.yaml
Comment (hoppea2) [inline bundle/manifests/dell-csm-operator.clusterserviceversion.yaml:5161]: Why are we downgrading min kube version?

File: config/manifests/bases/dell-csm-operator.clusterserviceversion.yaml
Comment (hoppea2) [inline config/manifests/bases/dell-csm-operator.clusterserviceversion.yaml:2444]: Why are we downgrading min kube version?

**Review by AkshaySainiDell — APPROVED** [review]

File: bundle/manifests/dell-csm-operator.clusterserviceversion.yaml
Comment (rajendraindukuri) [reply bundle/manifests/dell-csm-operator.clusterserviceversion.yaml:17]: Hi @hoppea2 As per the info we got sometime back, it is mentioned that they are going to stay this release.. I don't have any idea if that changed later.

File: bundle/manifests/dell-csm-operator.clusterserviceversion.yaml
Comment (rajendraindukuri) [reply bundle/manifests/dell-csm-operator.clusterserviceversion.yaml:5161]: @rajendraindukuri I think minkuberversion is updated based on the manifests generated based on the latest operator-sdk. In our case, we need to mention the lowest kubernetes version supported across all orchestrators we are claiming support. Based on that ... the latest MKE/RKE version we have in support matrix is on k8s 1.24.. Thats the reason we are updating this to 1.24 from last few releases manually.

**Comment by JacobGros** [comment]
some of the images referenced in this PR aren't pushed yet; IMO it would make more sense to wait for every component to release first, and then we can run some e2e on this branch before merging this PR and releasing operator

File: operatorconfig/moduleconfig/replication/v1.10.0/controller.yaml
Comment (santhoshatdell) [inline operatorconfig/moduleconfig/replication/v1.10.0/controller.yaml:275]: It would be better if the _imagePullPolicy_ is set to 'IfNotPresent' for versioned images.
This applies to other components as well.
CC @donatwork

**Review by anandrajak1 — APPROVED** [review]

File: operatorconfig/moduleconfig/replication/v1.10.0/controller.yaml
Comment (JacobGros) [reply operatorconfig/moduleconfig/replication/v1.10.0/controller.yaml:275]: I agree, but it is too late to make more changes. We can tackle that first thing in CSM 1.13

**Review by hoppea2 — APPROVED** [review]
LGTM

**Review by jooseppi-luna — APPROVED** [review]
lgtm

**Pull request merged**
